### PR TITLE
Discussion: Channel Import Data Format

### PR DESCRIPTION
**Pull request by dirkkelly**
[**Migration PR#56**](https://github.com/breadtubetv/breadtubetv/pull/56/files)

Hi All!

James has done absolutely incredible work putting together the proper architecture and framework our main clip executable, and it's written in Go! 

The first script we're tackling is importing channels, and this is raising questions about the data structure and what should be imported and what should be left for custom configuration.

I'm opening this PR as a way to review how the data is changing from running the command, a test-ground for iterating on the data structure and subsequent layouts until we have a version which we want agree on.

This is a discussion for #51 which has the results from `2018 Jan 25` of the command 

`go run main.go channel import youtube https://www.youtube.com/channel/UCEWrkBTDSYqJzQmp8gxQiMw`

My goal here is to keep the meta data that's relevant to the channel and only replace content for the provider, which in the first case is YouTube. That metadata includes slugs which could differ from what comes out of YouTube (some YouTube channels also don't have slugs).

To get everyone up to speed I'm going to go over how the data is structured in master, where it's going in the current development branch #51, and what I think it needs to be in order to keep data that's not from the YouTube API.

Master Data Structure `data/channels/dirkkelly.yml`

```yaml
name: "Dirk Kelly"
slug: "dirkkelly"
url: "https://www.youtube.com/dirkkelly"
description: "Austramerican thinking about politics, economics, and technology during the people's struggle to survive the collapse of capitalism."
subscribers: 12
tags: ["breadtube"]
```

Current In Development Structure

```yaml
name: Dirk Kelly
slug: dirkkelly
url: https://www.youtube.com/dirkkelly
subscribers: 12
tags:
- breadtube
```

Proposed Finalized Structure (additional attributes for illustration, can be added later)

```yaml
name: "Dirk Kelly"
slug: "dirkkelly"
added: "2018-01-19"
updated: "2018-01-25"
description: "Gardening, Programming, and Video Games from an Australian Anarchists Perspective"
providers:
  youtube:
    name: "Dirk Kelly"
    id: "UC6YURf7CpM2hH1AxxbelTrg"
    url: "https://www.youtube.com/dirkkelly"
    slug: "dirkkelly"
    description: "Austramerican thinking about politics, economics, and technology during the people's struggle to survive the collapse of capitalism."
    joined: "2011-10-03"
    subscribers: 12
    views: 855
tags:
- "breadtube"
```

Running an Import would be about changing the provider data, not the surrounding data, so the script will need to get executed with the BreadTube slug, which isn't the youtube slug, this way you know which record to update, eg:

`go run main.go channel import dirkkelly youtube https://www.youtube.com/channel/UCEWrkBTDSYqJzQmp8gxQiMw`

### CODE REVIEW

File: data/channels/dirk-kelly.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name: Dirk Kelly
+slug: dirk-kelly
```
Comment (dirkkelly) [inline data/channels/dirk-kelly.yml]: This is why we're going to need to import based on the slug, so it knows which channel data file to add/update.

File: data/channels/democracyatwork.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: Democracy At Work
+slug: democracyatwork
+url: https://www.youtube.com/democracyatwrk
```
Comment (dirkkelly) [inline data/channels/democracyatwork.yml]: another example, the url doesn't have a nice slug, ours can

**Comment by dirkkelly** [comment]
This branch is not generating a preview because it is not a PR onto master.

**Comment by Murodese** [comment]
Proposed changes lgtm. As long as the provider stuff is in a series of maps I can structuremap it to yaml.

**Comment by dirkkelly** [comment]
Added support in #56

**Pull request closed without merging**